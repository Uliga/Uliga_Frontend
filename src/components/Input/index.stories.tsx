import React from "react";
import { withKnobs } from "@storybook/addon-knobs";
import Input from "./index";

export default {
  title: "components/Input", // 스토리북에서 보여질 그룹과 경로를 명시
  component: Input, // 어떤 컴포넌트를 문서화 할지 명시
  decorators: [withKnobs], // 애드온 적용
};

export function DefaultInput() {
  return <Input value="TEST" labelHidden size={100} />;
}
export function CheckBox() {
  return <Input type="checkbox" value="" />;
}
export function LabelInput() {
  return <Input size={30} value="label" label="label" />;
}
export function ControlTest(args: any) {
  return <Input {...args} />;
}
ControlTest.args = {
  value: "test",
  size: "30rem",
};
