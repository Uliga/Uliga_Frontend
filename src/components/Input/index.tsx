import React from "react";
import styled from "styled-components";
import COLORS from "../../constants/color";

const Wrapper = styled.div``;
const Label = styled.div`
  font-size: 1.3rem;
  color: ${COLORS.GREY[500]};
  padding-bottom: 1rem;
`;

const InputWrapper = styled.input<{ size: number }>`
  ${({ size }) => `
    width: ${size}rem;
`}
  padding: 1.5rem 1.2rem;
  border-radius: 0.5rem;
  border: 0.1rem solid ${COLORS.GREY[300]};
  color: ${COLORS.GREY[500]};
  font-size: 1.3rem;
  ::placeholder {
    color: ${COLORS.GREY[400]};
    font-weight: 300;
  }
`;

const Message = styled.p`
  font-size: 1.1rem;
  padding-top: 1rem;
  color: ${COLORS.MEDIUM_BLUE};
`;

type InputProps = {
  /** input 안의 초깃값 */
  value: string;
  /** 변화가 일어났을떄 호출되는 함수 */
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  /** input type : ex> text,password,checkbox etc. */
  type?: string;
  /** 자리표시자 */
  placeholder?: string;
  /** label 숨길지 여부 */
  labelHidden?: boolean;
  /** label 내용 */
  label?: string;
  /** input 너비 */
  size?: number;
  /** 필수로 채워야하는 필드 */
  required?: boolean;
  /** 읽기 전용 */
  readOnly?: boolean;
  /** input의 입력 조건 */
  message?: string;
  className?: string;
};

export default function Input({
  value,
  onChange,
  type,
  placeholder = "",
  labelHidden = false,
  label,
  size = 35,
  required = false,
  className,
  readOnly = false,
  message = "",
}: InputProps) {
  return (
    <Wrapper className={className}>
      {!labelHidden && <Label>{label}</Label>}
      <InputWrapper
        value={value}
        onChange={onChange}
        type={type}
        placeholder={placeholder}
        size={size}
        required={required}
        readOnly={readOnly}
      />
      {message.length > 0 && <Message> {message}</Message>}
    </Wrapper>
  );
}
