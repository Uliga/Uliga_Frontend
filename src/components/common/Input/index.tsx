import React from "react";
import styled from "styled-components";
import { GREY } from "../../../constants/color";

const Wrapper = styled.div``;
const Label = styled.div`
  font-size: 1.3rem;
  color: ${GREY[500]};
  padding-bottom: 1rem;
`;

const InputWrapper = styled.input<{
  size: number;
}>`
  ${({ size }) => `
    width: ${size}rem;
`}
  padding: 1.5rem 1.2rem;
  border-radius: 0.5rem;
  border: 0.1rem solid ${GREY[300]};
  color: ${GREY[500]};
  font-size: 1.3rem;
  ::placeholder {
    color: ${GREY[400]};
  }
`;

type InputProps = {
  /** input 안의 초깃값 */
  value?: string;
  /** 변화가 일어났을떄 호출되는 함수 */
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  /** input type : ex> text,password,checkbox etc. */
  type?: string;
  /** 자리표시자 */
  placeholder?: string;
  /** label 유무 */
  labelExist?: boolean;
  /** label 내용 */
  label?: string;
  /** input 너비 */
  size?: number;
  /** 필수로 채워야하는 필드 */
  required?: boolean;
  className?: string;
};

export default function Input({
  value,
  onChange,
  type,
  placeholder = "내용을 입력해주세요.",
  labelExist,
  label,
  size = 35,
  required = false,
  className,
}: InputProps) {
  return (
    <Wrapper className={className}>
      {labelExist && <Label>{label}</Label>}
      <InputWrapper
        value={value}
        onChange={onChange}
        type={type}
        placeholder={placeholder}
        size={size}
        required={required}
      />
    </Wrapper>
  );
}
