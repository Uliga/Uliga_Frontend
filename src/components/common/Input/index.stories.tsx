import React from "react";
import { withKnobs } from "@storybook/addon-knobs";
import Input from "./index";

export default {
  title: "components|basic/Input", // 스토리북에서 보여질 그룹과 경로를 명시
  component: Input, // 어떤 컴포넌트를 문서화 할지 명시
  decorators: [withKnobs], // 애드온 적용
};

export function Input2() {
  return <Input size={30} labelExist label="내용" />;
}
