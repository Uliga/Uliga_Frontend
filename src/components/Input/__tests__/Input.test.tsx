import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Input from "../index";

describe("<Input />", () => {
  it("Input 컴포넌트 스냅샷 테스트", () => {
    const inputValue = "value";
    const testUtils = render(<Input value={inputValue} />);
    expect(testUtils.container).toMatchSnapshot();
  });
  it("Input 컴포넌트가 dom에 존재하는지 확인한다.", () => {
    const inputValue = "value";
    const { getByDisplayValue } = render(<Input value={inputValue} />);
    const inputComponent = getByDisplayValue(inputValue);
    expect(inputComponent).toBeInTheDocument();
  });
  it("Input 컴포넌트 라벨이 dom에 있는지 확인한다.", () => {
    const inputValue = "value";
    const inputLabel = "label";
    const { getByText } = render(
      <Input value={inputValue} label={inputLabel} />,
    );
    const labelElement = getByText(inputLabel);
    expect(labelElement).toBeInTheDocument();
  });
  it("Input 컴포넌트의 placeholder 텍스트가 dom에 있는지 확인한다.", () => {
    const placeholderText = "placeholder";
    const { getByPlaceholderText } = render(
      <Input value="" placeholder={placeholderText} />,
    );
    const inputComponent = getByPlaceholderText(placeholderText);
    expect(inputComponent).toBeInTheDocument();
  });
  it("Input 컴포넌트의 하단 message가 dom에 있는지 확인한다.", () => {
    const messageText = "message";
    const { getByText } = render(<Input value="" message={messageText} />);
    const messageElement = getByText(messageText);
    expect(messageElement).toBeInTheDocument();
  });
});
