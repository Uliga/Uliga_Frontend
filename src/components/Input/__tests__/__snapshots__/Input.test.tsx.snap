// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Input 컴포넌트 스냅샷 테스트 1`] = `
<div>
  <div
    class="sc-bcXHqe"
  >
    <div
      class="sc-gswNZR cdSGNK"
    />
    <input
      class="sc-dkrFOg dyctdc"
      placeholder=""
      size="35"
      value="value"
    />
  </div>
</div>
`;
